packages/flutter/test/material/flat_button_test.dart

```diff
@@ -882,6 +883,54 @@ void main() {
       await buildWidget(buttonMinWidth: buttonMinWidth);
       expect(tester.widget<RawMaterialButton>(rawMaterialButtonFinder).constraints.minWidth, buttonMinWidth);
     });
+
+    testWidgets(
+    'Adaptive FlatButton displays CupertinoButton in iOS',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Material(
+              child: FlatButton.adaptive(
+                child: Text("Button"),
+                onPressed: () {},
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(find.byType(CupertinoButton), findsOneWidget);
+    }, variant: const TargetPlatformVariant(<TargetPlatform> {
+      TargetPlatform.iOS,
+      TargetPlatform.macOS,
+    })
+  );
+
+  testWidgets(
+    'Adaptive FlatButton does not display CupertinoButton in non-iOS',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Material(
+              child: FlatButton.adaptive(
+                child: Text("Button"),
+                onPressed: () {},
+              ),
+            ),
+          ),
+        ),
+      );
+
+      expect(find.byType(CupertinoButton), findsNothing);
+    }, variant: const TargetPlatformVariant(<TargetPlatform> {
+      TargetPlatform.android,
+      TargetPlatform.fuchsia,
+      TargetPlatform.windows,
+      TargetPlatform.linux,
+    }),
+  );
 }
```

Comment:
we should add some tests that the various constructors in the adaptive constructor actually translate into the cupertino button's attributes.